src/base/utils/io.cpp

```diff
@@ -89,13 +89,34 @@ nonstd::expected<QByteArray, Utils::IO::ReadError> Utils::IO::readFile(const Pat
         return nonstd::make_unexpected(ReadError {ReadError::ExceedSize, message});
     }
 
-    // Do not use `QIODevice::readAll()` it won't stop when reading `/dev/zero`
-    const QByteArray data = file.read(fileSize);
-    if (const qint64 dataSize = data.size(); dataSize != fileSize)
+    const qint64 targetSize = (maxSize < 0) ? fileSize : std::min(fileSize, maxSize);
```

Comment:
Isn't file size checked above and it is knowingly less or equal to max size at this point?

Reply:
Sorry, what do you imply?

Reply:
> Sorry, what do you imply?

Just use `fileSize` below.

Reply:
Comments are addressed.